src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -127,6 +127,13 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
           val dataFrameName = outputConfig.dataFrameName
           val dataFrame = repartition(outputConfig, job.sparkSession.table(dataFrameName))
 
+          outputConfig.outputOptions.get("protectFromEmptyOutput").asInstanceOf[Option[Boolean]] match {
```

Comment:
```suggestion
          (outputConfig.outputOptions.get("protectFromEmptyOutput"), dataFrame.isEmpty) match {
            case (Some(true), true)  =>
              throw MetorikkuWriteFailedException(s"Failed to write dataframe: ${dataFrameName}, " + s"empty output is not allowed according configuration")
            case _=>
          }
```

Reply:
OMG, how didn't I know it was possible!!!!
The end to nested cases